HW2/P2/P2.py

```diff
@@ -21,31 +23,40 @@
 
     total = orig_counts.sum()
 
-    # serial move
-    counts = orig_counts.copy()
-    with Timer() as t:
-        move_data_serial(counts, src, dest, 100)
-    assert counts.sum() == total, "Wrong total after move_data_serial"
-    print("Serial uncorrelated: {} seconds".format(t.interval))
-    serial_counts = counts.copy()
-
-    # fine grained
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_fine_grained(counts, src, dest, 100)
-    assert counts.sum() == total, "Wrong total after move_data_fine_grained"
-    print("Fine grained uncorrelated: {} seconds".format(t.interval))
+    t_serial = [0]*3
+    t_fine = [0]*3
+    t_medium = np.array([0.0]*21)
+    t_medium = t_medium.reshape(3,7)
 
-    ########################################
-    # You should explore different values for the number of locks in the medium
-    # grained locking
-    ########################################
-    N = 10
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained uncorrelated: {} seconds".format(t.interval))
+    for i in range(3):
+        # serial move
+        counts = orig_counts.copy()
+        with Timer() as t:
+            move_data_serial(counts, src, dest, 100)
+        assert counts.sum() == total, "Wrong total after move_data_serial"
+        print("Serial uncorrelated: {} seconds".format(t.interval))
+        serial_counts = counts.copy()
+        t_serial[i] = t.interval
+
+        # fine grained
+        counts[:] = orig_counts
+        with Timer() as t:
+            move_data_fine_grained(counts, src, dest, 100)
+        assert counts.sum() == total, "Wrong total after move_data_fine_grained"
+        print("Fine grained uncorrelated: {} seconds".format(t.interval))
+        t_fine[i] = t.interval
+
+        ########################################
+        # You should explore different values for the number of locks in the medium
+        # grained locking
+        ########################################
+        for N in [j*2 for j in range(1,8)]:
```

Comment:
Yep, did similar things to you. One thing to note is that you can accomplish the for loop you have typed here in a slightly more "pythonic" way by doing for N in range(1, 8, 2) <- skips by 2.